module/hooks/json-changes.mjs

```diff
@@ -15,12 +15,31 @@ const cullKeys = {
   //packs: ['private']
 };
 
+const deprecatedKeys = {
+  packs: {
+    entity: "type", // v9 deprecation (entity -> type), v10 incompatible
+  }
+};
+
+/**
+ * @param {Object} data
+ * @param {Object} result
+ */
+function preScrubData(data, result) {
+  if (data.systems) {
+    data.system = data.systems;
+    delete data.systems;
+    result.details.deprecations.systems = "system";
+  }
+}
+
 /**
  * @param {String} file File to fetch
- * @param {Object} current Currently loaded data.
+ * @param {SystemData|ModuleData} loadedPkg Currently loaded data.
  * @param {Function} comparer Custom comparer instead of default diffObject
  */
-function fetchData(file, current, comparer) {
+function fetchData(file, loadedPkg, comparer) {
+  loadedPkg = foundry.utils.deepClone(loadedPkg);
```

Comment:
instead of reassigning the input variable, I'd prefer assigning a new one with the changed object
```suggestion
function fetchData(file, current, comparer) {
  const loadedPkg = foundry.utils.deepClone(current);
```